pkg/cmd/dcos.go

```diff
@@ -32,10 +33,33 @@ const annotationUsageOptions string = "usage_options"
 // NewDCOSCommand creates the `dcos` command with its `auth`, `config`, and `cluster` subcommands.
 func NewDCOSCommand(ctx api.Context) *cobra.Command {
 	cmd := &cobra.Command{
-		Use: "dcos",
+		Use:  "dcos",
+		Args: cobra.ArbitraryArgs,
 		PersistentPreRun: func(cmd *cobra.Command, args []string) {
 			cmd.SilenceUsage = true
 		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return cmd.Help()
+			}
+			switch args[0] {
+			case "job", "marathon", "node", "package", "service", "task":
```

Comment:
Is there a reason why we don't extract the plugin during the setup and remove the overhead when running those commands the first time?

Reply:
The plugin is installed during setup, this is only for the scenario when a user updates their CLI and they have a previously configured cluster (without the core-cli plugin).